frontend/src/metabase/common/components/QuestionPicker/components/QuestionPicker.tsx

```diff
@@ -38,6 +39,7 @@ interface QuestionPickerProps {
   options: QuestionPickerOptions;
   path: QuestionPickerStatePath | undefined;
   shouldShowItem?: (item: QuestionPickerItem) => boolean;
+  onInit: (item: QuestionPickerItem) => void;
```

Comment:
Do we need it? When `onItemSelect` doesn't work?

Reply:
We need to differentiate what is being auto-selected from what is being user-selected because there's an instance of  `CollectionPickerModal` where `hasConfirmButtons` is `false`, which means it automatically submits the form when `onItemSelect` is called - we don't want to automatically submit the form when the initial first collection is automatically selected (prod requirement).

I tried to add an extra param to `onItemSelect` to pass this information but it made `onItemSelect` handlers quite complex. A dedicated prop made everything simpler (actually I could not get it to work reliably with a flag in `onItemSelect`).